# CSE110 Assignment 2 

This assignment is to guide you to solve basic programming problems using loop in Python. 

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**

## Write the Python code of the following problems:

### Task 1

Write python program, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18,-27, 36,-45,54,-63

=====================================================================

**Hints(1):** Use a while loop for solving these problems.

**Hints(2):** We are already familiar with the print() function. But when we use it to print any value it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we need to add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter should be initialized at 24 and the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24-18=6. So the loop counter value is getting decremented by 6 in every iteration.

For your understanding task 1(a) code is done for you.


In [ ]:
# a) 24, 18, 12, 6, 0, -6

# initialize loop counter
counter = 24

# loop structure
while counter >= -6:
    
    #inside loop body
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 
     
    counter = counter - 6 #updating loop counter
    #inside loop body
    
#outside loop body

24, 18, 12, 6, 0, -6

In [ ]:
# b)-10, -5, 0, 5, 10, 15, 20
for counter in range(-10,21,5):
    if counter == 20:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 

-10, -5, 0, 5, 10, 15, 20

In [ ]:
# c)18, 27, 36, 45, 54, 63
for counter in range(18,64,9):
    if counter == 63:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 

18, 27, 36, 45, 54, 63

**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [2]:
# d)18,-27, 36,-45,54,-63
index = 1;
for counter in range( 18, 64, 9):
    
    
    if index%2 != 0:
          print(counter* 1, end = ", ") 
          index+=1
    else:
          print(counter* -1, end = ", ") 
          index+=1

    

18, -27, 36, -45, 54, -63, 

====================================================================\
<font color='blue'>
All the tasks can be solved using both the "while loop" and the "for loop". But you need to choose the best looping construct which is the most suitable for solving a particular task.
</font>

====================================================================

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

**Example 01:** If the user enters “Toyota” and 20, your program should print the name “Toyota” twenty times.

**Input:**\
Toyota\
2

**Output:**\
Toyota\
Toyota

==========================================================

**Example02:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Input:**\
Veyron\
5

**Output:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [ ]:
#Todo
carName = input("Enter a car name: ")
repeat = int(input("Enter a number: "))
for r in range(repeat):
  print(carName)

hukkahua
3
hukkahua
hukkahua
hukkahua


### Task 3
Write a Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600).

Output: 2835


In [ ]:
#Todo
sum = 0
for r in range(601):
  if r%7==0 and r%9==0:
    sum += r
print(sum)

2835


### Task 4
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600(including 600). \
Ensure that numbers like 63 are added only once in the sum.

Output: 42649


In [ ]:
#Todo
sum = 0
for r in range(601):
  if r%7==0 or r%9==0:
    sum += r
print(sum)

42649


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600(including 600).

Output: 39814


In [ ]:
#Todo
sum = 0
for r in range(601):
  if r%7==0 and r%9==0:
    continue
  elif r%7==0 or r%9==0:
    sum += r
print(sum)

39814


### Task 6
Write a Python code to displays all the **odd numbers** between 10 and 50.

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [3]:
#Todo
sum = 0
for r in range(11,50,2):
  print(r,end=" ")

11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $


==========================================================

**Example01:** 

**Input:**\
10

**Output:**\
-55

==========================================================

**Example02:** 

**Input:**\
20

**Output:**\
-210



In [4]:
#Todo
n = int(input("Enter a number: "))
sum = 0
for r in range(1,n+1):
  if r%2==0:
    sum -= r**2
  else:
    sum += r**2
print(sum)

Enter a number: 10
-55


### Task 8
Write a Python code of a program that asks the user to enter ten numbers then display the total and the average of **ONLY** the **odd numbers** among those ten numbers. <font color='red'>[Use of list is completely unnecesary for this task]</font>
==========================================================

**Example01:** 

**Input:**\
1

2

3

4

5

6

7

8

9

10

**Output:**
Total is 25 and Average is 5.0


**Explanation:**

Total is 25 = (1+3+5+7+9) and Average is 25/5 = 5.0

==========================================================


**Example02:** 

**Input:**\
-20

3

-5

40

-17

9

20

-8

99
  
-200

**Output:**

Total is 89 and Average is 17.8


**Explanation:**

Total is 89 =(3+(-5)+(-17)+9+99) and Average is 89/5 = 17.8



In [ ]:
#Todo
sum = 0
for r in range(10):
  num = int(input("Enter a number: "))
  if num%2 != 0:
    sum += num
print("Total is "+str(sum)+" and Average is "+str(sum/5))

1
2
3
4
5
6
7
8
9
10
Total is 25 and the Average is 5.0


### Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

==========================================================

**Example01:** 

**Input:**\
50

**Output:**\
196

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 = 196

==========================================================

**Example02:** 

**Input:**\
75

**Output:**\
385

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385





In [6]:
#Todo
sum = 0
num = int(input("Input a number 'N': "))
for r in range(1, num+1):
  if r%7 == 0:
    sum += r

  
  
print(sum)

Input a number 'N': 50
196


### Task 10
Write a Python code that will read 10 numbers from the user, and then print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 10 numbers.

==========================================================


For example,

The user enters 10, output 10, then \
The user enters 2, (10+2) = 12, output 12, then \
The user enters 4, (10+2+4) = 16, output 16, then \
The user enters 20, (10+2+4+20) = 36, output 36, and continuous till the 10$^{th}$ input


In [ ]:
#Todo
sum = 0

for r in range(10):
  num = int(input("Enter a number: "))
  sum+=num
  print(sum)

  
  

1
1
1
2
1
3
1
4
1
5
1
6
1
7
1
8
1
9
1
10


### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

<font color='red'>[Consider the input number to be an INTEGER]</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

=========================================================================

**Hint:**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.



In [7]:
#Todo
num = input("Enter a number: ")
for r in range(-1,-len(num)-1,-1):
  if r == (-len(num)):
    print(num[r],end="")
  else:
    print(num[r],end=",")


Enter a number: 32768
8,6,7,2,3

### Task 12
Write a Python program that takes a number from the user and prints its digits from left to right. <font color='red'> (left to right) </font>

<font color='red'>[The input number has to be an INTEGER]</font>

=========================================================================

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

=========================================================================

**Hint:**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.


In [ ]:
#Todo
num = input("Enter a number: ")
for r in range(len(num)):
  if r == (len(num)-1):
    print(num[r],end="")
  else:
    print(num[r],end=",")

123
1,2,3

### Task 13

Write a Python program that takes a number and tells how many digits are in that number. \
<font color='red'>[Consider the input number has to be an INTEGER] </font>

<font color='red'>You are not allowed to use len() function</font>

**Example:** If the user gives 9876, you should print 4. 



**Hint:** Keep floor dividing by ten and count how many times this could be divided. 

9876 floor divide by 10, is 987, count that got 1 digit (total 1) \
987 floor divide by 10, is 98, count that got 1 digit (total 2) \
98 floor divide by 10, is 9, count that got 1 digit (total 3) \
9 floor divide by 10, is 0, count that got 1 digit (total 4) 

Done! (When the number becomes 0 your loop should end.)



In [ ]:
#Todo
num = int(input("Enter a number to know how many digits are there in that number: "))
count = 0
while num%10!=0:
  count+= 1
  num = num//10
print(count)

9876
4


### Task 14

Write a Python program that takes a number from the user and prints the divisors of that number and then print how many divisors were there.
<font color='red'>[The input number has to be an INTEGER]</font>

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
1, 2, 3, 6 \
Total 4 divisors.


==========================================================

**Example02:** 

**Input:**\
121

**Output:**\
1, 11, 121 \
Total 3 divisors.




In [ ]:
#Todo
num = int(input("Enter a number to check its devisors : "))
count = 0
for r in range(1,num+1):
    if (num%r==0):
      if (r != num):
        print(r,end = ",")
      else:
        print(r)
      count+=1
print("Total "+str(count)+" devisors")

Enter a number to check its devisors : 6
1,2,3,6
Total 4 devisors


### Task 15
Write a Python program that takes a number and tells if it is a perfect number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
6 is a perfect number

**Explanation:**\
6 have 4 divisors: 1, 2, 3, and 6. \
If we add all factors except itself, 6 = 1 + 2 + 3.

==========================================================


**Example02:** 

**Input:**\
20

**Output:**\
28 is a perfect number

**Explanation:**\
28 have 6 divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself, 28 = 1 + 2 + 4 + 7 + 14.

==========================================================

**Example03:** 

**Input:**\
33

**Output:**\
33 is not a perfect number

**Explanation:**\
33 have 3 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11.



In [8]:
#Todo
num = int(input("Enter a number to check if its perfect or not: "))
count = 0
for r in range(1,num+1):
    if (num%r==0 and r != num):
      count+=r
      
if (count == num):
  print(str(count)+' is a perfect number')
else:
  print(str(count)+' is not a perfect number')



Enter a number to check if its perfect or not: 28
28 is a perfect number


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Prime Number:** If a number has only two divisors, (1 and itself), then it is prime. If it is divisible by more numbers, then it is not a prime.

==========================================================

**Hint:** use the divisor count from task 16. 

==========================================================

**Example01:** 

**Input:**\
11

**Output:**\
11 is a prime number

**Explanation:**\
11 has only 2 divisors: 1, and 11.

==========================================================

**Example02:** 

**Input:**\
6

**Output:**\
6 is not a prime number

**Explanation:**\
6 have 4 divisors: 1, 2, 3 and 6.






In [ ]:
#Todo
#Todo
num = int(input("Enter a number to check if its prime or not: "))
count = 0
for r in range(1,num+1):
    if (num%r==0):
     
      count+=1
if count >2:
  print(str(num)+" is not a prime number.")
else:
  print(str(num)+" is a prime number.")


121
121 is not a prime number.


### Task 17

Write a Python program that asks the user for a quantity then takes that many numbers and prints the maximum, minimum and average of those numbers. 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS][FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

==========================================================

**Example01:** If the user enters 5 as an input for quantity. Then enters 10, 4, -1, -100, and 1. \
Your program output should be: “Maximum 10”, “Minimum -100”, “Average is -17.2”

**Input:**\
5
10
4
-1
-100
1.

**Output:**\
Maximum 10\
Minimum -100\
Average is -17.2

**Explanation:**\
Average calculation: (10+4 + (-1) + (-100) + 1)/5 = -86/5=-17.2



In [ ]:
#Todo
many = int(input("How many numbers do you want to work with: "))
sum = 0
min = 999999999999999999999999999999999
max =-999999999999999999999999999999999

for r in range(many):
  num = int(input("Enter a number of your choice: "))
  
  sum += num
  if num>max:
    max = num
  if num<min:
    min = num
print("Maximum "+str(max))
print("Minimum "+str(min))
print("Average is "+ str(sum/many))

How many numbers do you want to work with: 3
Enter a number of your choice: 1
Enter a number of your choice: 2
Enter a number of your choice: 3
Maximum 3
Minimum 1
Average is 2.0


## Task 18

Write a python program that prints a square of size N using + where N will be given as input.

=======================================

**Sample Input**
5

**Sample Output**<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>


=======================================

**Sample Input**
3

**Sample Output**<br>
+++<br>
+++<br>
+++<br>

In [ ]:
#Todo
num = int(input("Enter number: "))
for r in range(num):
  for r in range(num):
    print("+",end="")
  print("")

Enter number: 5
+++++
+++++
+++++
+++++
+++++


### Task 19

Write a python program that prints a rectangle of size M (height/line numbers) and N(length/column numbers) using incrementing numbers where M,N will be given as input.

===================================

**Sample Input**
4,6

**Sample Output**<br>
123456<br>
123456<br>
123456<br>
123456<br>


===================================


**Sample Input**
3,2

**Sample Output**<br>
12<br>
12<br>
12<br>


In [9]:
#Todo
num = input("Enter two number separated by ','. eg.2,3 : ")
for r in range(int( num[0])):
  for r in range(int( num[2])):
    print(str(r+1),end="")
  print("")

Enter two number separated by ','. eg.2,3 : 4,5
12345
12345
12345
12345


### Task 20

Write a python program that prints a right angled triangle of height N using incrementing numbers where N will be given as input.

====================================

**Sample Input**
4

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>

====================================

**Sample Input**
5

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>
12345<br>

In [ ]:
#Todo
num = int(input("Enter number: "))
for r in range(1,num+1):
  for n in range(1,r+1):
    print(n,end="")
  print("")

Enter number: 6
1
12
123
1234
12345
123456


## Optional Tasks (21 -26) [Ungraded]

### Task 21

Write a python program that prints all the fibonacci number from 0 to N where N will be given. 

**A Fibonacci number is a number which is the summation of its previous two fibonacci number.**

First two fibonacci number are 0 and 1. So the 3rd Fib will be 0+1=1, 4th Fib will be 1+1=2, 5th Fib will be 1+2=3 and so on.

==============================

**Sample Input**<br>
10

**Sample Output**<br>
0 1 1 2 3 5 8

==============================


**Sample Input**<br>
15

**Sample Output**<br>
0 1 1 2 3 5 8 13

In [10]:
#Todo
num = int(input(""))
current = 0
previous = 0
buffer = 0
count = 0
while current < num and previous < num:
  
  if count == 0:
    print(0,end=" ")
  elif count ==1:
    print(1,end=" ")
  elif count ==2:
    
    previous = 1
    print(1,end=" ")
    current = 2
  else:
    print(current,end=" ")
    buffer = previous
    previous = current
    current = previous + buffer
    
  count+=1
  


15
0 1 1 2 3 5 8 13 

### Task 22

Write a python program that converts a Decimal Integer number to a Boolean Number. 

**A decimal can be converted to a binary number by keeping track of the remainders after each division of that number by 2. **

================================

For example, to convert 10 to a binary number, we can follow the following approach

10/2 = 5 (Remainder 0) 
<br>5/2 = 2 (Remainder 1) 
<br>2/2 = 1 (Remainder 0) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1010. Binary of 10 is 1010.

**Sample Input** <br>
10

**Sample Output**<br>
1010

================================

For example, to convert 13 to a binary number, we can follow the following approach

13/2 = 6 (Remainder 1) 
<br>6/2 = 3 (Remainder 0) 
<br>3/2 = 1 (Remainder 1) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1101. Binary of 13 is 1101.

**Sample Input** <br>
13

**Sample Output**<br>
1101

In [ ]:
#Todo
qution = ""
remainder = 0
num = int(input("Enter a number:"))
arr = []
while qution !=0:
  qution = num//2
  remainder = num%2
  arr.append(remainder)
  num =  qution
for r in range(len(arr)-1,-1,-1):
  print(arr[r],end = "")

Enter a number:13
1101

### Task 23

Write a python program that converts a Binary number to a Decimal Integer Number. 

**A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2 where the power denotes the position of the binary digit**

========================================

For example, to convert 1010, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 0 + $2^{1}$ x 1 + $2^{0}$ x 0 = 10


**Sample Input** <br>
1010

**Sample Output** <br>
10

========================================

For example, to convert 1101, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 1 + $2^{1}$ x 0 + $2^{0}$ x 1 = 13


**Sample Input** <br>
1101

**Sample Output** <br>
13

In [ ]:
#Todo

#authentication
num = input("Enter a binary number:")
isBinary = False
while isBinary == False:
  isBinary = True
  for r in num:
    if  (r =="0" or r=="1"):
      pass
    else:
      isBinary = False
  if (isBinary == False):
    num = input("Enter a binary number:")
  
#calculating the denary
sum = 0
count =0
for r in range(len(num)-1,-1,-1):
  # print("r="+str(r))
  # print("arr[r]="+str(num[r]))

  sum += int(num[r])*(2**count)
  count+=1
# print(num)
print(sum)

Enter a binary number:1010
10


### Task 24

Take five numbers from the user and find the minimum and the average of **only the even** numbers entered by the user. [If the user enters odd numbers ignore them] 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS] [FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

Assume, the first input is always an even number.

============================================================

**Example01:** If the user enters 10, 4, -1, -100, and 1. \
Output: “Minimum -100”, “Average is 28.66667”

**Input:**\
10

4

-1

-100

1

**Output:**\
Minimum -100\
Average is 28.66667

**Explanation:**\
Average calculation: (10+4 + (-100))/3 = -86/3=-28.66667

============================================================

**Example02:** If the user enters 2, 10, 1, 21, and 3. \
Output: Minimum 2”, “Average is 6.0”

**Input:**\
2

10

1

21

3

**Output:**\
Minimum 2\
Average is 6.0

**Explanation:**\
Average calculation: (2+10)/2 = 12/2= 6





In [ ]:
#Todo
sum = 0
count = 0
min = 999999999999999999999999999999999999999999999999999999999999999999999999999999999999999

for r in range(5):
  num = int(input("Enter a number: "))
  if num%2 == 0:
    sum += num
    count += 1
  if min>num:
    min = num
if count ==0:
  count =1
print("Minimum "+str(min))
print("Average is "+str(sum/count))

Enter a number: 5
Enter a number: 6
Enter a number: 6
Enter a number: 6
Enter a number: 6
Minimum 5
Average is 6.0


### Task 25

Write a Python code for the following:
* Ask the user to enter a number, N (Total number of inputs to be taken). Assume that the user will never enter the first number as zero.
* Take N number of inputs
* Prints out the product of all the numbers read

For example, if the first input is 4, then the program has to read in four numbers from the user and print the product of these four numbers. 

=========================================================================

**Example01:**

**Input:**\
5 

10 

6 

4 

2 

1 

**Output:**\
480

**Explanation:**\
5 is the total number of inputs taken. Then, the calculation should be 10 X 6 X 4 X 2 X 1 = 480 







In [ ]:
#Todo
product = 1
n = int(input("enter a number: "))

for r in range(n):
  num = int(input("Enter a number: "))
  product *= num
print(product)


enter a number: 5
Enter a number: 6
Enter a number: 1
Enter a number: 1
Enter a number: 1
Enter a number: 1
6


### Task 26
Ask the user for a range (a starting number and an ending number). 
**Count** how many numbers are prime numbers and how many numbers are perfect numbers between that range and **print** those numbers. 


=========================================================================

**Hint (1):** Declare two strings/lists to store the prime and perfect numbers. Inside the iteration store the values of the prime and perfect numbers in the pre- declared variables.

**Hint (2):**
Need to use string concat.

ans = "Prime numbers:"\
prime_value = 11

ans = ans + str(11)

print(ans)

ans = ans + str(13) 

print(ans)


=========================================================================

**Example:**between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6). 

**Input:**\
2

6

**Output:**\
Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number.\
Prime numbers: 2, 3, 5.\
Perfect numbers: 6.



In [ ]:
#Todo
num1 = int(input("enter a number: "))
num2 = int(input("enter another number: "))
strPerfect = ""
strPrime = ""
for r in range(num1,num2+1,1):
  num = r
  countPerfect = 0
  for r in range(1,num+1):
      if (num%r==0 and r != num):
        countPerfect+=r
        
  if (countPerfect == num):
    
    if len(strPerfect)>=1:
      strPerfect += ", "+str(r)
    else:
      
      strPerfect += str(r)
  


  num = r
  countPrime = 0
  for r in range(1,num+1):
      if (num%r==0):
      
        countPrime+=1
  if countPrime <=2:
    if len(strPrime)>=1:
      strPrime += ", "+str(r)
    else:
      strPrime += str(r)
  
# Between 2 and 6,
# Found 3 prime numbers
# Found 1 perfect number.
# Prime numbers: 2, 3, 5.
# Perfect numbers: 6.
print("Between "+str(num1)+" and "+str(num2)+",")
print("Found "+str(countPrime)+" prime numbers")
print("Found "+str(countPerfect)+" perfect number.")
print("Prime numbers: "+strPrime)
print("Perfect numbers: "+strPerfect)


enter a number: 1
enter another number: 100
0
Between 1 and 100,
Found 9 prime numbers
Found 117 perfect number.
Prime numbers: 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97
Perfect numbers: 6, 28
